### ISSUE TITLE
 [CodePush] javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7e22c1d0:

### ISSUE THREAD

**Issue by zhazhangjun** [Microsoft/react-native-code-push#1546]
Thanks so much for filing an issue or feature request! Please fill out the following (wherever relevant):

### Steps to Reproduce
03-18 13:18:05.046 17551-17578/? I/ReactNativeJS: [CodePush] Reporting binary update (1.2.4)
03-18 13:18:06.516 17551-17578/? I/ReactNativeJS: '--checkForUpdate--', '\'英文说明+++德文说明+++中文说明\'', 20, '{"deploymentKey":"l4QoO11Y8qYzGoLDSQfSfkxnSU_d8083febc-5fa5-4c39-8402-71dfa4879dd7","description":"\'英文说明+++德文说明+++中文说明\'","label":"v6","appVersion":"1.2.4","isMandatory":false,"packageHash":"e3338a1ab935f588f669b6d79ca68ea30bdb530fadddfe7e45f5f21de0b9a47c","packageSize":5418421,"downloadUrl":"https://codepushupdates.azureedge.net/storagev2/73X3RPHm7PRzHgmS-Z1Wgif-qW6k656808dd-2cbb-4f8d-b450-7625ce684632","isPending":false,"failedInstall":false}'
03-18 13:18:06.526 17551-17578/? I/ReactNativeJS: Checking for update.
03-18 13:18:06.806 17551-17578/? I/ReactNativeJS: Awaiting user action.
03-18 13:18:07.996 17551-17578/? I/ReactNativeJS: ===delayedFailed==Not Found! Params: {"ret":null,"key":"introF","autoSync":false,"syncInBackground":true}
03-18 13:18:08.196 17551-17578/? I/ReactNativeJS: ===FirstIntro=language=err=[object Object]
03-18 13:18:26.706 17551-17578/? I/ReactNativeJS: Downloading package.
03-18 13:18:28.376 17551-17578/? I/ReactNativeJS: An unknown error occurred.
    [CodePush] javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7e22c1d0: Failure in SSL library, usually a protocol error
    error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:744 0x72275d74:0x00000000)

* react-native-code-push version: v2.1.9

* react-native version:react-native-cli: 2.0.1
                                 react-native: 0.51.0

**Comment by alexandergoncharov** [comment]
Hi @zhazhangjun ,
Thanks for reporting!

Could you please share your CodePush settings from JS code part?

**Comment by bhansen77** [comment]
I am receiving this problem as well - no changes made, but this has started happening 

03-19 11:33:46.990: W/System.err(2201): javax.net.ssl.SSLHandshakeException: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7c1502a8: Failure in SSL library, usually a protocol error
03-19 11:33:46.990: W/System.err(2201): error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:741 0x735dfcf8:0x00000000)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:448)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.Connection.upgradeToTls(Connection.java:146)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.Connection.connect(Connection.java:107)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:294)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpEngine.sendSocketRequest(HttpEngine.java:255)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:206)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:345)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:296)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getHeaderField(HttpURLConnectionImpl.java:143)
03-19 11:33:46.990: W/System.err(2201): 	at java.net.URLConnection.getHeaderFieldInt(URLConnection.java:544)
03-19 11:33:46.990: W/System.err(2201): 	at java.net.URLConnection.getContentLength(URLConnection.java:316)
03-19 11:33:46.990: W/System.err(2201): 	at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getContentLength(HttpsURLConnectionImpl.java:182)
03-19 11:33:46.990: W/System.err(2201): 	at com.microsoft.codepush.react.CodePushUpdateManager.downloadPackage(CodePushUpdateManager.java:166)
03-19 11:33:46.990: W/System.err(2201): 	at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:234)
03-19 11:33:46.990: W/System.err(2201): 	at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:228)
03-19 11:33:46.990: W/System.err(2201): 	at android.os.AsyncTask$2.call(AsyncTask.java:288)
03-19 11:33:46.990: W/System.err(2201): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
03-19 11:33:46.990: W/System.err(2201): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
03-19 11:33:46.990: W/System.err(2201): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
03-19 11:33:47.000: W/System.err(2201): 	at java.lang.Thread.run(Thread.java:841)
03-19 11:33:47.000: W/System.err(2201): Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7c1502a8: Failure in SSL library, usually a protocol error
03-19 11:33:47.000: W/System.err(2201): error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:741 0x735dfcf8:0x00000000)
03-19 11:33:47.000: W/System.err(2201): 	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
03-19 11:33:47.000: W/System.err(2201): 	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:405)
03-19 11:33:47.000: W/System.err(2201): 	... 19 more
03-19 11:33:47.010: I/ReactNativeJS(2201): [CodePush] An unknown error occurred.
03-19 11:33:47.010: I/ReactNativeJS(2201): [CodePush] javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7c1502a8: Failure in SSL library, usually a protocol error
03-19 11:33:47.010: I/ReactNativeJS(2201): error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:741 0x735dfcf8:0x00000000)

**Comment by zt5** [comment]
android 4.x same error reason [https://developer.android.com/about/versions/android-5.0-changes#ssl](url) 
solve  [https://blog.csdn.net/yanzhenjie1003/article/details/80202476](url)

**Comment by bhansen77** [comment]
The devices attempting to connect are indeed android 4.x. Based on the above comment I tested a 5.x device and found that it worked.

This only started happening recently, previous to that android 4.x worked fine. The devices that I have using this are all 4.x and cannot be upgraded. Can support for 4.x be restored?

**Comment by Jeijie** [comment]
any updates? I have same issue on Android 4.4.4

**Comment by zhazhangjun** [comment]
Hi：@alexandergoncharov  
This is my CodePush settings from JS code part
syncImmediate1() {
        CodePush.checkForUpdate(CODE_PUSH_KEY).then((update) => {
        // CodePush.checkForUpdate().then((update) => {

           this.updateDes= update.description;
            console.log('--checkForUpdate--',update.description, this.updateDes.length, JSON.stringify(update));
            if (!update) {

                Toast.show(<Text style={{color: 'white', textAlign: 'center'}}> 已经最新版本 </Text>, '3', null, false);
            } else {
                // this.setState({modalVisible: true, updateInfo: update, isMandatory: update.isMandatory});
                if(update.description.length>5){
                    this.showSync();
                }else{
                    this.silentSync();
                }
            }
        });

    }

    showSync(){
        CodePush.sync(
            {
                deploymentKey: CODE_PUSH_KEY,
                installMode: CodePush.InstallMode.IMMEDIATE,
                // installMode: CodePush.InstallMode.ON_NEXT_RESUME,
                updateDialog : {
                    //是否显示更新描述
                    appendReleaseDescription : true ,
                    //更新描述的前缀。 默认为"Description"
                    // descriptionPrefix : <Text style={{color:'white'}}>更新内容</Text>,
                    // descriptionPrefix : "\n \nDescrption：\n1、This is descriptionPre1123\n2、descriptionPre2\n\n" ,
                    descriptionPrefix : "\n \n更新内容：\n" ,
                    //强制更新按钮文字，默认为continue
                    mandatoryContinueButtonLabel : "立即更新" ,
                    //强制更新时的信息. 默认为"An update is available that must be installed."
                    mandatoryUpdateMessage : "必须更新后才能使用",
                    // mandatoryUpdateMessage : "An update is available that must be installed.",   //"必须更新后才能使用" ,
                    //非强制更新时，按钮文字,默认为"ignore"
                    optionalIgnoreButtonLabel : '稍后' ,
                    //非强制更新时，确认按钮文字. 默认为"Install"
                    optionalInstallButtonLabel : '极速更新' ,
                    //非强制更新时，检查到更新的消息文本
                    optionalUpdateMessage : '有新版本了，是否更新？' ,
                    //Alert窗口的标题
                    title : '更新提示'
                } ,

            },
            this.codePushStatusDidChange.bind(this),
            this.codePushDownloadDidProgress.bind(this)
        )
    }

**Comment by zhazhangjun** [comment]
No at present

 

发件人: Jeijie [mailto:notifications@github.com] 
发送时间: 2019年3月21日 星期四 11:16
收件人: Microsoft/react-native-code-push
抄送: zhazhangjun; Mention
主题: Re: [Microsoft/react-native-code-push] [CodePush] javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x7e22c1d0: (#1546)

 

any updates? I have same issue on Android 4.4.4

**Comment by yuri-kulikov** [comment]
Hi everyone, you can check my [message](https://github.com/Microsoft/react-native-code-push/issues/1551#issuecomment-475672654) from a similar issue.

We have found a workaround and are working on a fix.

**Comment by esutton** [comment]
Same issue on Android 4.4.4 KitKat
````
    "react-native-code-push": "^5.6.0",
    "react": "16.3.1",
    "react-native": "0.55.4",
````

adb logcat
````
[CodePush] Exception
javax.net.ssl.SSLHandshakeException: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0xb9c5ac90: Failure in SSL library, usually a protocol error
error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:741 0x96f99926:0x00000000)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:448)
	at com.android.okhttp.Connection.upgradeToTls(Connection.java:146)
	at com.android.okhttp.Connection.connect(Connection.java:107)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:294)
	at com.android.okhttp.internal.http.HttpEngine.sendSocketRequest(HttpEngine.java:255)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:206)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:345)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:296)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:179)
	at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:246)
	at com.microsoft.codepush.react.CodePushUpdateManager.downloadPackage(CodePushUpdateManager.java:167)
	at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:217)
	at com.microsoft.codepush.react.CodePushNativeModule$3.doInBackground(CodePushNativeModule.java:211)
	at android.os.AsyncTask$2.call(AsyncTask.java:288)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:841)
Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0xb9c5ac90: Failure in SSL library, usually a protocol error
error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:741 0x96f99926:0x00000000)
	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:405)
	... 17 more
````

**Comment by yuri-kulikov** [comment]
Hey everyone! I know, this issue isn't solved yet, but it's the same issue as #1551. I'll close this issue so as not to divide the discussion on this. We'll share all the new information about this in #1551!
